### DISCUSSION TITLE
How to solve IsADirectoryError: [Errno 21] Is a directory: 'tgcf.config.yml' when doing docker run

### DISCUSSION THREAD

**Discussion by manojsitapara** [aahnik/tgcf#187] [Q&A] [answered]
**Describe the bug**
When running docker container with following command, it throws following error

```shell
INFO:telethon.crypto.libssl:Failed to load SSL library: <class 'OSError'> (no library called "ssl" found)
INFO:root:tgcf.config.yml detected.
Traceback (most recent call last):
  File "/venv/bin/tgcf", line 8, in <module>
    sys.exit(app())
  File "/venv/lib/python3.9/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "/venv/lib/python3.9/site-packages/tgcf/cli.py", line 81, in main
    from tgcf.live import start_sync  # pylint: disable=import-outside-toplevel
  File "/venv/lib/python3.9/site-packages/tgcf/live.py", line 9, in <module>
    from tgcf import config, const
  File "/venv/lib/python3.9/site-packages/tgcf/config.py", line 148, in <module>
    CONFIG = read_config()
  File "/venv/lib/python3.9/site-packages/tgcf/config.py", line 94, in read_config
    with open(CONFIG_FILE_NAME) as file:
IsADirectoryError: [Errno 21] Is a directory: 'tgcf.config.yml'
```

I tried

```shell
docker run -v tgcf.config.yml:/app/tgcf.config.yml -d --env-file .env aahnik/tgcf
```

**System information:**
Windows

After google this error, I tried following command, it started running but not forwarding messages

`docker run -v tgcf.config.yml:/app/ -d --env-file .env aahnik/tgcf:minimal`

**Comment by aahnik** [comment, marked as answer]
The docker method is for people who know docker.



**As you are a windows user, it's best that you follow [this guide](https://github.com/aahnik/tgcf/wiki/Run-tgcf-on-Windows)**

------

**The solution to the mistake you made**

You must use an **absolute file path** for docker volumes. 

`/absolute/path/to/tgcf.config.yml` is just an example in the [docker guide](https://github.com/aahnik/tgcf/wiki/Install-and-run-using-docker#run). It must be replaced with the actual path of the file on your computer. 

As you have just written just `tgcf.config.yml`, docker thinks that it is a folder. An empty folder at the root (which does not exist) is mounted in the working dir of the docker container.

Learn
- For windows, the path separator is `\`. For Linux and Mac, it is `/`
- [Absolute vs relative path](https://linuxhandbook.com/absolute-vs-relative-path/)
- [Docker Volumes](https://docs.docker.com/storage/volumes/)
    The `-v` flag specifies volume to mount. The syntax is `-v <path on host>:<path on container>`

**Reply by aahnik** [reply]
@manojsitapara please star this repo, and share with your friends/social media.

For updates, please "Watch" the repo on github. Also you may join the telegram channel [aahniks_code](https://telegram.me/aahniks_code)

**Reply by manojsitapara** [reply]
Thanks I did star this repo.
